[Mohamed Hamas] iP

Bahamas20 opened the pull request:
# DukeSIU
> "By failing to prepare, you are preparing to fail." - Benjamin Franklin [source](https://www.goodreads.com/quotes/15061-by-failing-to-prepare-you-are-preparing-to-fail#:~:text=Quote%20by%20Benjamin%20Franklin%3A%20%E2%80%9CBy,you%20are%20preparing%20to%20fail.%E2%80%9D)

DukeSIU helps you to prepare by remembering the things you have to do, It's,

- Easy to use
- Convenient 
- Super fast to use 

All you need to do is,

1. Download
2. Run DukeSIU
3. Add your tasks
4. Let it handle the rest :star_struck:

AND IT IS **FREE** SIUUU

Features:
- [x] Managing Tasks
- [x] Managing Deadlines
- [x] Managing Events 
- [ ] Reminders (coming soon)

Here's the main method: 

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Bahamas20 committed:
Level 1

Bahamas20 committed:
Level 2

Bahamas20 committed:
Level 3

Bahamas20 committed:
Level 3

Bahamas20 committed:
Level 4

Bahamas20 committed:
Level 6

Bahamas20 committed:
Level 6 v2

Bahamas20 committed:
Lvl 6 final

Bahamas20 committed:
A-text

Bahamas20 committed:
A-TextUiTesting

Bahamas20 committed:
Minor edits

Bahamas20 committed:
minor

Bahamas20 committed:
Level-7

Bahamas20 committed:
Level-8

Bahamas20 committed:
A-MoreOOP

Bahamas20 committed:
A-Packages

Bahamas20 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'
Gradle

Bahamas20 committed:
A-Gradle

Bahamas20 committed:
A-JUnit

Bahamas20 committed:
test

Bahamas20 committed:
test1

Bahamas20 committed:
Merge branch 'test'

* test:
  test1

Bahamas20 committed:
Javacdoc for deadline

Bahamas20 committed:
Javadoc for deadline

Bahamas20 committed:
test
1

Bahamas20 committed:
A-JavaDoc

Bahamas20 committed:
A-CodingStandard

Bahamas20 committed:
Level-9

Bahamas20 committed:
merge

Bahamas20 committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  A-JavaDoc

# Conflicts:
#	src/main/java/duke/TaskList.java

Bahamas20 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  A-CodingStandard

# Conflicts:
#	src/main/java/duke/Duke.java

Bahamas20 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Level-9

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

normkoh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,81 @@
+            } else if (first.equals("list")) {
+                tLst.list();
+            } else if (first.equals("mark")) {
+                Task current = tLst.get(Integer.parseInt(str.split(" ")[1]) - 1);
```
I feel like this line may be a bit difficult to read and understand what it does. 
Perhaps you could consider storing the `str.split(" ")[1]) - 1)` part in a separate variable first, since it also appears multiple times in subsequent lines.

normkoh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,81 @@
+                    if (str.endsWith("deadline")) {
+                        throw new DukeException("Oops The description of deadline cannot be empty !");
+                    }
+                    String currD = str.substring(str.indexOf("deadline") + 8, str.indexOf("/by"));
+                    tLst.add(new Deadline(currD.substring(1), str.substring(str.indexOf("/by") + 4)));
```
Maybe you could consider breaking the logic up in these two lines of code into smaller, separate lines of code? 

I also found it a little bit difficult to figure out what the 8 and 4 represents. Maybe creating and assigning them to constants might help to add meaning to these arbitrary numbers.

normkoh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,81 @@
+                    if (str.endsWith("event")) {
+                        throw new DukeException("Oops The description of event cannot be empty !");
+                    }
+                    String currE = str.substring(str.indexOf("event") + 5, str.indexOf("/at"));
+                    tLst.add(new Events(currE.substring(1), str.substring(str.indexOf("/at") + 4)));
```
Same comment as lines 55-56.

normkoh reviewed (commented):
Overall, in terms of code quality, I found that it was pretty well done. There are just some individual lines where I feel breaking up the logic would help in readability and understanding of the code.  Other than that, I feel that the code quality is quite solid. Great job!

Bahamas20 committed:
Update README.md

Bahamas20 committed:
Update README.md

Bahamas20 committed:
Update README.md

Bahamas20 committed:
Update README.md

waynezsy commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,42 @@
+ */
+public class Deadline extends Task {
+    protected String by;
+    public Deadline(String description,String by) {
```
Be careful and remember to add a white space after the comma in your constructor arguments :)

waynezsy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,48 @@
+            tasks = new TaskList();
+        }
+    }
+        public void run() throws DukeException, IOException {
```
Indentation error?
Noticed the same issue on line 45.

waynezsy commented on src/main/java/duke/Events.java:
```diff
@@ -0,0 +1,25 @@
+ */
+public class Events extends Task {
+    protected String at;
+    public Events(String description,String at) {
```
Remember to add a white space after commas :)

waynezsy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,102 @@
+ * where data represents the list of tasks
+ * to be done
+ */
+
+
```
I think the coding style guide says there should not be white spaces between the JavaDoc and the class/method.

waynezsy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,102 @@
+    Storage(String filePath) {
+        
+        this.filePath = filePath;
+        this.file = new File(filePath);
+        this.tasks = new ArrayList<>();
```
I don't think there's a need for the "this" on these 2 lines since the fields aren't shadowed by a constructor parameter? (Based on the advanced conventions)

waynezsy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,102 @@
+        catch (FileNotFoundException e) {
+            this.file = new File(filePath);
+
+    } catch (DukeException e) {
```
I think this line is indented wrongly

waynezsy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,102 @@
+                    }
+                }
+            }
+        }
+        catch (FileNotFoundException e) {
```
I think the coding style enforces that catch should be on the same line as the bracket

waynezsy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,102 @@
+
+        String textToAdd = "";
+        for (Task item : tasks) {
+            if (item != null)
+                textToAdd += tasks.indexOf(item) + 1 + "." + item + "\n";
```
I think the coding style insists on brackets for the if body even though there is no else

waynezsy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,105 @@
+        lst = new ArrayList<>();
+    }
+
+    /**
+     * Returns the task in specified index
+     * @param ind index of task
+     * @return Task at index ind
+     */
```
I realise many of your JavaDoc share this coding style issue:
There needs to be a whitespace between the description and the @ statements. I think fullstops (or punctuation) are also required

waynezsy commented on src/main/java/duke/ToDos.java:
```diff
@@ -0,0 +1,23 @@
+/**
+ * A task class to indicate a ToDo task,
+ * only requires a description of task
+ *
```
Probably don't need this empty line?

waynezsy commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,32 @@
+ * Represents the User interface,
+ * used to interact with the user
+ */
+
```
I think this empty line might need to be removed?

waynezsy reviewed (commented):
Overall your code follows the coding style conventions quite closely, though I think you made a few careless mistakes here and there. Perhaps you can consider adding doing the A-CheckStyle increment even though it's optional as it makes it really a lot easier to ensure your code fits with the coding standard :)

Bahamas20 committed:
Level-10

Bahamas20 committed:
no message

Bahamas20 committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Level-10

# Conflicts:
#	src/main/java/duke/Duke.java

Bahamas20 committed:
Merge branch 'master' of https://github.com/Bahamas20/ip

* 'master' of https://github.com/Bahamas20/ip:
  Update README.md
  Update README.md
  Update README.md
  Update README.md

Bahamas20 committed:
some clean up

Bahamas20 committed:
Add assertions

No assertions presents

Assertions need to be added to define assumptions so that runtime can verify

Assertions being added to different classes

Done this way to check if any input is null as it should not be

Bahamas20 committed:
A-CodeQuality

Bahamas20 committed:
Updated code to improve code quality

For better readability and structure

Following the coding quality standards

Bahamas20 committed:
Merge pull request #2 from Bahamas20/branch-A-Assertions

A-Assertions

Bahamas20 committed:
Merge branch 'master' of https://github.com/Bahamas20/ip

* 'master' of https://github.com/Bahamas20/ip:
  Add assertions

Bahamas20 committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add assertions

# Conflicts:
#	src/main/java/duke/Storage.java

Bahamas20 committed:
Merge branch 'master' into branch-A-Assertions

* master:

Bahamas20 committed:
Merge pull request #3 from Bahamas20/branch-A-CodeQuality

A-CodeQuality

Bahamas20 committed:
Merge branch 'master' of https://github.com/Bahamas20/ip

* 'master' of https://github.com/Bahamas20/ip:
  Updated code to improve code quality
  A-CodeQuality

Bahamas20 committed:
Merge branch 'master' into branch-A-CodeQuality

* master:

Bahamas20 committed:
Create gradle.yml

Bahamas20 committed:
A-CI

Bahamas20 committed:
Merge branch 'master' of https://github.com/Bahamas20/ip

* 'master' of https://github.com/Bahamas20/ip:
  A-CI
  Create gradle.yml

Bahamas20 committed:
BCD-Extension

Bahamas20 committed:
Merge branch 'branch-BCD-Extension'

* branch-BCD-Extension:
  BCD-Extension

Bahamas20 committed:
A-Personality

Bahamas20 committed:
Merge branch 'master' into branch-A-Assertions

* master:
  A-Personality
  BCD-Extension
  A-CI
  Create gradle.yml
  Updated code to improve code quality
  A-CodeQuality

Bahamas20 committed:
Merge branch 'branch-A-Assertions' into branch-A-CodeQuality

* branch-A-Assertions:
  A-Personality
  BCD-Extension
  A-CI
  Create gradle.yml

Bahamas20 committed:
no message

Bahamas20 committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:

Bahamas20 committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:

Bahamas20 committed:
A-BetterGui

Bahamas20 committed:
Website

Bahamas20 committed:
Update README.md

Bahamas20 committed:
Update README.md

Bahamas20 committed:
Update README.md

Bahamas20 committed:
Update README.md

Bahamas20 committed:
Update README.md

Bahamas20 committed:
Update README.md

Bahamas20 committed:
Update README.md

Bahamas20 committed:
Update README.md

Bahamas20 committed:
Update README.md

Bahamas20 committed:
Merge branch 'master' of https://github.com/Bahamas20/ip

* 'master' of https://github.com/Bahamas20/ip:
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md

Bahamas20 committed:
no message

Bahamas20 committed:
A-Release

Bahamas20 committed:
Final changes to ip

Bahamas20 committed:
Final changed after peer review
